superset-frontend/src/components/TimezoneSelector/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { useChangeEffect } from 'src/common/hooks/useChangeEffect';
+
+const DEFAULT_TIMEZONE = 'GMT Standard Time';
+
+const offsetsToName = {
+  '-300-240': ['Eastern Standard Time', 'Eastern Daylight Time'],
+  '-360-300': ['Central Standard Time', 'Central Daylight Time'],
+  '-420-360': ['Mountain Standard Time', 'Mountain Daylight Time'],
+  '-420-420': [
+    'Mountain Standard Time - Phoenix',
+    'Mountain Standard Time - Phoenix',
+  ],
+  '-480-420': ['Pacific Standard Time', 'Pacific Daylight Time'],
+  '-540-480': ['Alaska Standard Time', 'Alaska Daylight Time'],
+  '-600-600': ['Hawaii Standard Time', 'Hawaii Daylight Time'],
+  '60120': ['Central European Time', 'Central European Daylight Time'],
+  '00': [DEFAULT_TIMEZONE, DEFAULT_TIMEZONE],
+  '060': ['GMT Standard Time - London', 'British Summer Time'],
+};
+
+const currentDate = moment();
+const JANUARY = moment([2021, 1]);
+const JULY = moment([2021, 7]);
+
+const getOffsetKey = (name: string) =>
+  JANUARY.tz(name).utcOffset().toString() +
+  JULY.tz(name).utcOffset().toString();
+
+const getTimezoneName = (name: string) => {
+  const offsets = getOffsetKey(name);
+  return (
+    (currentDate.tz(name).isDST()
+      ? offsetsToName[offsets]?.[1]
+      : offsetsToName[offsets]?.[0]) || name
+  );
+};
+
+export interface TimezoneProps {
+  onTimezoneChange: (value: string) => void;
+  timezone?: string | null;
+}
+
+const ALL_ZONES = moment.tz
+  .countries()
+  .map(country => moment.tz.zonesForCountry(country, true))
+  .flat();
+
+const TIMEZONES: moment.MomentZoneOffset[] = [];
+ALL_ZONES.forEach(zone => {
+  if (
+    !TIMEZONES.find(
+      option => getOffsetKey(option.name) === getOffsetKey(zone.name),
+    )
+  ) {
+    TIMEZONES.push(zone); // dedupe zones by offsets
+  }
+});
+
+const TIMEZONE_OPTIONS = TIMEZONES.sort(
+  // sort by offset
+  (a, b) =>
+    moment.tz(currentDate, a.name).utcOffset() -
+    moment.tz(currentDate, b.name).utcOffset(),
+).map(zone => ({
+  label: `GMT ${moment
+    .tz(currentDate, zone.name)
+    .format('Z')} (${getTimezoneName(zone.name)})`,
+  value: zone.name,
+  offsets: getOffsetKey(zone.name),
+}));
+
+const timezoneOptions = TIMEZONE_OPTIONS.map(option => (
+  <Select.Option key={option.value} value={option.value}>
+    {option.label}
+  </Select.Option>
+));
+
+const TimezoneSelector = ({ onTimezoneChange, timezone }: TimezoneProps) => {
+  const matchTimezoneToOptions = (timezone: string) =>
+    TIMEZONE_OPTIONS.find(option => option.offsets === getOffsetKey(timezone))
+      ?.value || DEFAULT_TIMEZONE;
+
+  const updateTimezone = (tz?: string | null) => {
+    onTimezoneChange(matchTimezoneToOptions(tz || moment.tz.guess()));
+  };
+
+  // update when the value changes from previous
+  useChangeEffect(timezone, () => {
+    updateTimezone(timezone);
+  });
+
+  // update on load
+  useEffect(() => {
+    updateTimezone(timezone);
+  }, []);
+
+  return (
+    <Select
+      css={{ width: '450px' }}
```

Comment:
Use a grid here?

Reply:
I was thinking that the parent component would probably set this. I think I might change this to a min-width for readability and leave the rest of the styling to the parent component.

Reply:
![TimezoneSelector_-_Interactive_Timezone_Selector_⋅_Storybook](https://user-images.githubusercontent.com/5186919/126575532-134aa76b-c970-45ba-aaeb-ae483f94b0d0.png)
This is the smallest the component should be to render the longest value.